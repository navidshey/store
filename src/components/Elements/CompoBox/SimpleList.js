import React from 'react';
import PropTypes from 'prop-types';

const SimpleList = ({ title, list, name }) => {
  return (
    <select name={name} className="jquery-select">
      <option>{title}</option>
      {list.map(item => {
        return (
          <option value={item.value} key={item.value}>
            {item.title}
          </option>
        );
      })}
    </select>
  );
};

SimpleList.propTypes = {
  title: PropTypes.string,
  name: PropTypes.string,
  list: PropTypes.array
};
export default SimpleList;
