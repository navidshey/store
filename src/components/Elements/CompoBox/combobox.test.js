import React from 'react'
import SimpleList from './SimpleList'
import renderer from 'react-test-renderer'

// display test
describe('the simplelist', () => {
  it('renderers as expected - snapshot', () => {
    const tree = renderer
      .create(<SimpleList title='test list' name='test_list' list={[{ value: '1', title: 'title 11 ' }, { value: '2', title: 'title 22' }]}></SimpleList>)
      .toJSON()

    console.log(tree)

    expect(tree).toMatchSnapshot()
  })
})
