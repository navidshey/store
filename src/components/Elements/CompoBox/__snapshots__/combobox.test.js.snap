// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the simplelist renderers as expected - snapshot 1`] = `
<select
  className="jquery-select"
  name="test_list"
>
  <option>
    test list
  </option>
  <option
    value="1"
  >
    title 11 
  </option>
  <option
    value="2"
  >
    title 22
  </option>
</select>
`;
